test/e2e/annotations/affinity.go

```diff
@@ -223,4 +223,46 @@ var _ = framework.IngressNginxDescribe("Annotations - Affinity/Sticky Sessions",
 		Expect(resp.Header.Get("Set-Cookie")).Should(ContainSubstring(fmt.Sprintf("Expires=%s", expected)))
 		Expect(resp.Header.Get("Set-Cookie")).Should(ContainSubstring("Max-Age=259200"))
 	})
+
+	It("should not set affinity across all server locations when using separate ingresses", func() {
+		f.UpdateNginxConfigMapData("error-log-level", "warn")
+		host := "cookie.foo.com"
+
+		// TODO: add `"nginx.ingress.kubernetes.io/session-cookie-path":  "/foo/bar"`
+		// after https://github.com/kubernetes/ingress-nginx/pull/3383/ is merged
+		annotations := map[string]string{
+			"nginx.ingress.kubernetes.io/affinity":  "cookie",
+			"nginx.ingress.kubernetes.io/use-regex": "true",
+		}
+		ing1 := framework.NewSingleIngress("ingress1", "/foo/bar.*", host, f.IngressController.Namespace, "http-svc", 80, &annotations)
+		f.EnsureIngress(ing1)
+
+		ing2 := framework.NewSingleIngress("ingress2", "/foo.*", host, f.IngressController.Namespace, "http-svc", 80, &map[string]string{})
+		f.EnsureIngress(ing2)
+
+		f.WaitForNginxServer(host,
+			func(server string) bool {
+				return strings.Contains(server, `location ~* "^/foo/bar.*"`) && strings.Contains(server, `location ~* "^/foo.*"`)
+			})
+
+		resp, _, errs := gorequest.New().
+			Get(f.IngressController.HTTPURL+"/foo").
+			Set("Host", host).
+			End()
+
+		Expect(errs).Should(BeEmpty())
+		Expect(resp.StatusCode).Should(Equal(http.StatusOK))
+		Expect(resp.Header.Get("Set-Cookie")).Should(Equal(""))
+
+		resp, _, errs = gorequest.New().
+			Get(f.IngressController.HTTPURL+"/foo/bar").
+			Set("Host", host).
+			End()
+
+		Expect(errs).Should(BeEmpty())
+		Expect(resp.StatusCode).Should(Equal(http.StatusOK))
+		// TODO: change expected substring to `session-cookie-path` value
```

Comment:
you could just use normal paths instead of regexp paths and get rid of https://github.com/kubernetes/ingress-nginx/pull/3383 dependency, no?